Adding config for having suffix for client ID for realtime consumer

eaugene opened the pull request:
**Context**
For the consuming source of Pinot this can be used to give more which specify like consuming server is having issue. 
As currently the info is _tableNameWithType,  streamTopic, _partitionGroupId which is going to be the same for all the replicas of the consuming real-time segment. 
Using this config can be used to provide more granular info which can used to speed up debugging.

**Tags** 
- new config

**Todo** 
- https://github.com/pinot-contrib/pinot-docs/pull/337

Jackie-Jiang commented on pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:
```diff
@@ -699,6 +699,10 @@ public static class Server {
 
+    public static final String CONFIG_OF_REALTIME_SEGMENT_CONSUMER_CLIENT_ID_SUFFIX =
+        "realtime.consumer.client.id.suffix";
+    public static final String DEFAULT_REALTIME_SEGMENT_CONSUMER_CLIENT_ID_SUFFIX = "";
```
We usually use `null` if there is no default setting for a config. You may directly call `_serverConfig.getProperty(CONFIG_OF_REALTIME_SEGMENT_CONSUMER_CLIENT_ID_SUFFIX);`

Jackie-Jiang commented on pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:
```diff
@@ -699,6 +699,10 @@ public static class Server {
     public static final String CONFIG_OF_LOGGER_ROOT_DIR = "pinot.server.logger.root.dir";
 
+    public static final String CONFIG_OF_REALTIME_SEGMENT_CONSUMER_CLIENT_ID_SUFFIX =
+        "realtime.consumer.client.id.suffix";
```
More concise since real-time is implicit for consumer
```suggestion
        "consumer.client.id.suffix";
```

Jackie-Jiang commented on pinot-spi/src/main/java/org/apache/pinot/spi/config/instance/InstanceDataManagerConfig.java:
```diff
@@ -39,6 +39,8 @@ public interface InstanceDataManagerConfig {
 
   String getSegmentStoreUri();
 
+  String getRealtimeSegmentConsumerClientIdSuffix();
```
```suggestion
  String getConsumerClientIdSuffix();
```

Jackie-Jiang reviewed (approved):
LGTM with minor comments

eaugene commented:
Thanks for the review and approval @Jackie-Jiang . I have addressed the comments . 
Can you help with review of config doc https://github.com/pinot-contrib/pinot-docs/pull/337 ?
